### PR TITLE
doc: update maintainers and codeowners

### PR DESCRIPTION
**Pull request by kalyazin**
## Changes

Update maintainer and codeowner lists.

## Reason

To keep it up to date.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this
  PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- ~~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- ~~[ ] All added/changed functionality is tested.~~
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by zulinx86 — APPROVED** [review]

File: MAINTAINERS.md [hunk 1]
```diff
@@ -2,13 +2,12 @@
...
-- Diana Popa <dpopa@amazon.com>
 - Egor Lazarchuk <yegorlz@amazon.co.uk>
-- Jonathan Woollett-Light <jcawl@amazon.com>
 - Marco Cali <xmarcalx@amazon.com>
 - Nikita Kalyazin <kalyazin@amazon.com>
 - Pablo Barbachano <pablob@amazon.com>
 - Patrick Roy <roypat@amazon.com>
 - Takahiro Itazuri <itazur@amazon.com>
+- Riccardo Mancini <mancio@amazon.com>
+- Jack Thomson <jackabt@amazon.com>
```
Comment (zulinx86) [inline MAINTAINERS.md:13]: nit: sort by dictionary order?

Comment (xmarcalx) [reply MAINTAINERS.md:13]: +1 on @zulinx86  comment.

**Review by xmarcalx — COMMENTED** [review]
LGTM, just a minor nit as pointed out already.

File: MAINTAINERS.md (hunk 1 above)
Comment (kalyazin) [reply MAINTAINERS.md]: thanks, done

**Review by bchalios — APPROVED** [review]

**Review by zulinx86 — APPROVED** [review]

**Review by xmarcalx — APPROVED** [review]
LGTM thanks for the changes Nikita!

**Review by pb8o — APPROVED** [review]

**Pull request merged**